main.nf

```diff
@@ -214,35 +201,35 @@ process bwa {
 
     module 'bioinfo-tools'
     module 'bwa'
-    module 'bamtools'
     module 'samtools'
     module 'BEDTools'
-
+    
     cpus 2
-    memory '16 GB'
-    time '120h'
-
-    publishDir "$results_path/bwa"
-
+    memory { 16.GB * task.attempt }
+    time { 120.h * task.attempt }
+    errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
+    maxRetries 3
+    maxErrors '-1'
+    
+    publishDir "${params.outdir}/bwa", mode: 'copy'
+    
     input:
     file (reads:'*') from trimmed_reads
     set val(prefix) from name_for_bwa
 
     output:
-    file '*.sort.bam' into bam_picard
-    file '*.sort.bam.bai' into results
-    file '*.sort.bed' into results
-
+    file '*.sorted.bam' into bam_picard
+    file '*.sorted.bam.bai'
+    file '*.sorted.bed'
+    stdout into bwa_logs
+    
+    script:
     """
-    bwa mem -M $genomebwa ${reads} > ${prefix}.sam
-    samtools view -bT $genomebwa ${prefix}.sam > ${prefix}.bam
-    samtools sort ${prefix}.bam ${prefix}.sort
-    samtools index ${prefix}.sort.bam
-    rm ${prefix}.sam
-    rm ${prefix}.bam
-    bedtools bamtobed -i ${prefix}.sort.bam | sort -k 1,1 -k 2,2n -k 3,3n -k 6,6 > ${prefix}.sort.bed
+    bwa mem -M $index $reads | samtools view -b > ${prefix}.bam
```

Comment:
Note @chuan-wang - important change here that needs testing.

Instead of creating a temporary `.sam` file and converting it to `.bam`, I'm piping the output to samtools to convert to `.bam` directly. This should be faster and save filespace.

_However_, one but I was confused about was why you were specifying `-T $genomebwa` in the conversion step. I don't usually do this, I think it's only needed when some SAM header info is missing, or other stuff. But need to be sure that it still works as I have it.